public/app/features/dashboard/components/PanelEditor/getVizualizationOptions.tsx

```diff
@@ -25,6 +35,11 @@ export function getVizualizationOptions(props: OptionPaneRenderProps): OptionsPa
     },
   };
 
+  plugin.optionsBuilder?.setRegistryChangeHandler(() => {
+    // Publishing options change event when the options registry changes
+    panel.events.publish(new PanelOptionsChangedEvent());
+  });
```

Comment:
🤮 - don't like it, yet it felt like a fairly straightforward way. Any other ideas?